src/transformers/integrations.py

```diff
@@ -1045,6 +1051,52 @@ def __del__(self):
             self._ml_flow.end_run()
 
 
+class DagsHubCallback(MLflowCallback):
+    """
+    A [`TrainerCallback`] that logs to [DagsHub](https://dagshub.com/).
+    """
+
+    def __init__(self):
+        super().__init__()
+        if not is_dagshub_available():
+            raise RuntimeError("DagsHubCallback requires dagshub to be installed. Run `pip install dagshub`.")
+        from glob import glob
+
+        from dagshub.upload import Repo
+
+        self.Repo = Repo
+        self.glob = glob
+
+    def setup(self, *args, **kwargs):
+        """
+        Setup the DagsHub's Logging integration.
+
+        Environment:
+            HF_DAGSHUB_LOG_ARTIFACTS (`str`, *optional*):
+                Whether to save the data and model artifacts for the experiment. Default to `False`.
+        """
+
+        self.model = args[2]
+        self.log_artifacts = os.getenv("HF_DAGSHUB_LOG_ARTIFACTS", "FALSE").upper() in ENV_VARS_TRUE_VALUES
+        self.name = os.getenv("HF_DAGSHUB_MODEL_NAME") or "main"
+        self.remote = os.getenv("MLFLOW_TRACKING_URI") or input("Please input a remote url.")
```

Comment:
We can't use input in a training script that may be launched in several processes. You should error if this env variable is not set.